pkg/operators/wasm/maps.go

```diff
@@ -213,3 +276,43 @@ func (i *wasmOperatorInstance) mapDelete(ctx context.Context, m wapi.Module, sta
 	}
 	stack[0] = 0
 }
+
+// mapRelease close the map and release the handle.
+// Params:
+// - stack[0]: Map handle
+// Return value:
+// - 0 on success, 1 on error
+func (i *wasmOperatorInstance) mapRelease(ctx context.Context, m wapi.Module, stack []uint64) {
+	mapHandle := wapi.DecodeU32(stack[0])
+
+	i.createdMapMutex.RLock()
+	_, ok := i.createdMap[mapHandle]
+	i.createdMapMutex.RUnlock()
+
+	if !ok {
+		i.logger.Warnf("mapRelease: map %d was not created by newMap() or was already closed", mapHandle)
+		stack[0] = 1
+		return
+	}
+
+	ebpfMap, ok := getHandle[*ebpf.Map](i, mapHandle)
+	if !ok {
+		stack[0] = 1
+		return
+	}
+
+	err := ebpfMap.Close()
+	if err != nil {
+		i.logger.Warnf("mapRelease: closing map: %v", err)
+		stack[0] = 1
```

Comment:
I wonder if `delHandle` and `i.createdMap` should still be cleaned up when the `close()` syscall returns an error. This would prevent a wasm module from calling `mapRelease` twice, and then `close` twice on the same file descriptor. And the manpage of close says it is a bug to call close twice in case of error, because the fd can be closed even when returning an error, so the fd number can be reallocated to something else.

Current versions of Linux seems to never return an error when calling `close()` on fds of type bpf map, so it is not a problem in practice. But if that changes, that would be a problem.